src/main/java/com/hazelcast/aws/impl/DescribeInstances.java

```diff
@@ -236,14 +241,51 @@ private void addFilters() {
     InputStream callService(String endpoint) throws Exception {
         String query = getRequestSigner().getCanonicalizedQueryString(attributes);
         URL url = new URL("https", endpoint, -1, "/?" + query);
+
         HttpURLConnection httpConnection = (HttpURLConnection) (url.openConnection());
         httpConnection.setRequestMethod(Constants.GET);
         httpConnection.setConnectTimeout((int) TimeUnit.SECONDS.toMillis(awsConfig.getConnectionTimeoutSeconds()));
         httpConnection.setDoOutput(false);
         httpConnection.connect();
+
+        checkNoAwsErrors(httpConnection);
+
         return httpConnection.getInputStream();
     }
 
+    // TODO: Consider using this check for other connections in the hazelcast-aws project.
+    // TODO: Refactor class to use DI and avoid visible for testing

```

Comment:
I suggest to create github issues instead of TODO. It is more visible to everyone...

Reply:
Ok, done.
#62
#63